src/main/perl/MD.pm

```diff
@@ -55,10 +56,11 @@ sub _initialize
 
     my $st = $config->getElement($path)->getTree;
     $path=~m!/([^/]+)$!;
-    $self->{devname} = $1;
+    $self->{devname} = unescape($1);
     $st->{raid_level} =~ m!(\d)$!;
     $self->{raid_level} = $1;
     $self->{stripe_size} = $st->{stripe_size};
+    $self->{metadata} = ($st->{metadata}) ? $st->{metadata} : "0.90";
```

Comment:
http://linux.die.net/man/8/mdadm suggests 1.2 should be the default not 0.90, but in fact it's also optional so if it's unset then use the tool default.